.github/workflows/cloud-cleanup.yaml

```diff
@@ -31,7 +31,13 @@ jobs:
       - name: Delete stale CI instances
         run: |
           yesterday=$(date --utc --iso-8601=seconds --date=yesterday)
-          stale_instances=$(aws ec2 describe-instances --query "Reservations[].Instances[?LaunchTime<=\`$yesterday\`][].InstanceId" --filters 'Name=tag:Name,Values=uaclient-ci-*' --output text)
+          current_aws_principal=$(aws sts get-caller-identity | jq -r '.UserId')
+          stale_instances=$(
```

Comment:
The script is also better because it's able to delete more resources and not only instances, OTOH the simplicity of this approach makes it very solid. I also like that here we use the native `LaunchTime`, without relying on the instance name format. What I'd ultimately like to do is grow a general purpose cleanup script, reusable across projects and maybe handling multiple clouds via pycloudlib, but for now I'd go with this simpler approach